### PR TITLE
Add validation for extra properties in column descriptors

### PR DESCRIPTION
**Pull request by lewish**
This is all strongly typed, I think we can convert some other code to this pattern now.

### CODE REVIEW

File: core/column_descriptors.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+          path: currentPath
+        })
+      ];
+    }
+    const extraFields = Object.keys(description).filter(
+      key => !IRecordDescriptorFields.includes(key)
+    );
+    if (extraFields.length > 0) {
+      throw new Error(
+        `Unknown property "${extraFields[0]}" in column descriptor. Allowed properties are: ${IRecordDescriptorFields}`
```
Comment (BenBirt) [inline core/column_descriptors.ts:38]: Dan and I just agreed that we wouldn't fix this... It feels quite un-JavaScripty to do this, are you sure? I guess I'm happy if you are, though if we are going to do it I think we should be consistent with all `config` fields, as it'll be confusing if users start to expect us to do that and we don't

Comment (lewish) [reply core/column_descriptors.ts:38]: I'm happy to take this all the way... Having just written some configs, getting no feedback about if I'm doing it right I found very frustrating, particularly without good documentation for this alpha. 
We definitely get a fair few support requests where people have typos. We could downgrade the errors to not fully throw, but put them in the graph instead.

Comment (lewish) [reply core/column_descriptors.ts]: Have taken it all the way

File: scripts/lint [hunk 2]
```diff
@@ -1,6 +1,3 @@
 #!/bin/bash
 
-set -e
-
-bazel build //:tslint
-./bazel-bin/tslint "ts/**/*.{ts,tsx}"
+bazel build @npm//tslint/bin:tslint && bazel-bin/external/npm/tslint/bin/tslint.sh --project .
```
Comment (BenBirt) [inline scripts/lint:3]: nit: newline at EOF

File: tests/core/core.spec.ts [hunk 3]
```diff
@@ -571,6 +572,25 @@ suite("@dataform/core", () => {
...
+            type: "table",
+            columns: {
+              dimension_column: {
+                unknownProperty: "unknown"
+              } as any
+            }
+          })
+          .query(_ => "select 1 as test");
+      } catch (e) {
+        return;
```
Comment (BenBirt) [inline tests/core/core.spec.ts:589]: please add an `expect` asserting something about the expected Error

**Pull request merged**